library/core/src/time.rs

```diff
@@ -450,6 +476,31 @@ impl Duration {
         }
     }
 
+    /// Saturating `Duration` addition. Computes `self + other`, returning [`Duration::MAX`]
+    /// if overflow occurred.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// #![feature(duration_saturating_ops)]
+    /// #![feature(duration_constants)]
+    /// use std::time::Duration;
+    ///
+    /// assert_eq!(Duration::new(0, 0).saturating_add(Duration::new(0, 1)), Duration::new(0, 1));
+    /// assert_eq!(Duration::new(1, 0).saturating_add(Duration::new(u64::MAX, 0)), Duration::MAX);
+    /// ```
+    #[unstable(feature = "duration_saturating_ops", issue = "none")]
+    #[inline]
+    #[rustc_const_unstable(feature = "duration_consts_2", issue = "72440")]
+    pub const fn saturating_add(self, rhs: Duration) -> Duration {
+        match self.checked_add(rhs) {
+            Some(res) => res,
+            None => Duration::MAX,
+        }
```

Comment:
```suggestion
        self.checked_add(rhs).unwrap_or(Duration::MAX)
```

Reply:
Is `unwrap_or` const? I wanted to unstably constify those new methods directly.

Reply:
Ah, good question. Would you be up for `const`ifing `unwrap_or` first? 😇

Reply:
https://github.com/rust-lang/rust/pull/76330 a quick try if it's possible.

Reply:
@shepmaster That won't work yet. `const fn cannot evaluate destructors`.

Reply:
I chose to keep the current code as-is to unstably constify the new methods directly.